### DISCUSSION TITLE
oauth login and error hook

### DISCUSSION THREAD

**Discussion by gkielwasser** [directus/directus#9695] [Archive]
Hi,

I'm trying to handle the following situation:
- account b exists (created with the default provider)
- public registration is activated

A user try to authenticate on the same account using an other provider than the default.

An UNKNOWN_EXCEPTION is thrown because the email has to be unique.

My issue is I cannot find the proper way to handle this situation using a hook.

Here is the hooks I have tried:

1- filter('auth.login')
Result: the error is thrown before anything can be done in the hook. Hook is not called.

2- filter('create.user')
Result: the error is thrown before. Uniqness is checked before the hook is trigger. Hook is not called.

3- filter('request.error')
Result: the error is thrown before. Hook is not called.

For the last one at least, I'm not sure why this kind of request error wouldn't be catched

Thanks for @aidenfoxx for starting the discussion on Discord.

Would anyone has any suggestions about that?

Thanks

**Comment by rijkvanzanten** [comment]
You are correct. The user profile is created/extracted in the OAuth driver _before_ the auth.login hook is fired:

https://github.com/directus/directus/blob/6e439db6b28ce5608bff05a07b0301a5abf401c1/api/src/services/authentication.ts#L78-L83

causing the error to be thrown before the auth login can handle it. There's no way around this currently. The only solution would be to add a new set of hooks into the core platform for the specific steps in the auth provider's drivers. Do you wanna open a new feature request for these new hooks?

**Reply by gkielwasser** [reply]
This kind of error handling seems quite important to consolidate account creation using public registration configuration.

Here is a new feature request, I'm not sure the github request is correctly flagged, feel free to tell me if it is not the case,

https://github.com/directus/directus/discussions/10948
